src/frontend/src/optimizer/plan_node/generic/log_scan.rs

```diff
@@ -0,0 +1,229 @@
+        self.output_col_idx
+            .iter()
+            .map(|&i| format!("{}.{}", self.table_name, self.get_table_columns()[i].name))
+            .collect()
+    }
+
+    pub(crate) fn column_names(&self) -> Vec<String> {
+        self.output_col_idx
+            .iter()
+            .map(|&i| self.get_table_columns()[i].name.clone())
+            .collect()
+    }
+
+    pub fn distribution_key(&self) -> Option<Vec<usize>> {
+        let tb_idx_to_op_idx = self
+            .output_col_idx
+            .iter()
+            .enumerate()
+            .map(|(op_idx, tb_idx)| (*tb_idx, op_idx))
+            .collect::<HashMap<_, _>>();
+        self.table_desc
+            .distribution_key
+            .iter()
+            .map(|&tb_idx| tb_idx_to_op_idx.get(&tb_idx).cloned())
+            .collect()
+    }
+
+    /// get the Mapping of columnIndex from internal column index to output column index
+    pub fn i2o_col_mapping(&self) -> ColIndexMapping {
+        ColIndexMapping::with_remaining_columns(
+            &self.output_col_idx,
+            self.get_table_columns().len(),
+        )
+    }
+
+    /// Get the ids of the output columns and primary key columns.
+    pub fn output_and_pk_column_ids(&self) -> Vec<ColumnId> {
+        let mut ids = self.output_column_ids();
+        for column_order in self.primary_key() {
+            let id = self.get_table_columns()[column_order.column_index].column_id;
+            if !ids.contains(&id) {
+                ids.push(id);
+            }
+        }
+        ids
+    }
+
+    /// Create a logical scan node for log table scan
+    pub(crate) fn new(
+        table_name: String,
+        output_col_idx: Vec<usize>,
+        table_desc: Rc<TableDesc>,
+        op_column: Rc<ColumnCatalog>,
+        ctx: OptimizerContextRef,
+        old_epoch: u64,
+        new_epoch: u64,
+    ) -> Self {
+        Self {
+            table_name,
+            output_col_idx,
+            table_desc,
+            op_column,
+            chunk_size: None,
+            ctx,
+            old_epoch,
+            new_epoch,
+        }
+    }
+
+    pub(crate) fn columns_pretty<'a>(&self, verbose: bool) -> Pretty<'a> {
+        Pretty::Array(
+            match verbose {
+                true => self.column_names_with_table_prefix(),
+                false => self.column_names(),
+            }
+            .into_iter()
+            .map(Pretty::from)
+            .collect(),
+        )
+    }
+
+    pub(crate) fn get_id_to_op_idx_mapping(
+        output_col_idx: &[usize],
+        columns: Vec<ColumnDesc>,
+    ) -> HashMap<ColumnId, usize> {
+        let mut id_to_op_idx = HashMap::new();
+        output_col_idx
+            .iter()
+            .enumerate()
+            .for_each(|(op_idx, tb_idx)| {
+                let col = &columns[*tb_idx];
+                id_to_op_idx.insert(col.column_id, op_idx);
+            });
+        id_to_op_idx
+    }
+}
+
+impl GenericPlanNode for LogScan {
```

Comment:
The trait is implemented only to call `new_batch_with_core` of `PlanBase<Batch>`, which simply calls the `fn ctx()` and `fn schema()`. This two methods is simple. We can call `new_batch` directly with the ctx and schema, and then we can avoid implement the trait and implement so many unnecessary unused methods on the trait, and then we can further remove some unused methods.